packages/plugin-browser-device/test/device.test.ts

```diff
@@ -457,5 +458,96 @@ describe('plugin: device', () => {
       expect(sessions).toHaveLength(1)
       expect(sessions[0].device.id).toBe(events[0].device.id)
     })
+
+    it('should not set device.id as user.id when collectUserIp=true', () => {
+      const client = new Client(
+        { apiKey: 'API_KEY_YEAH' },
+        {
+          ...schema,
+          collectUserIp: {
+            defaultValue: () => true,
+            validate: () => true,
+            message: ''
+          }
+        },
+        [plugin(navigator)]
+      )
+      const events: EventWithInternals[] = []
+      const sessions: SessionWithDevice[] = []
+
+      expect(client._cbs.e).toHaveLength(1)
+
+      mockDelivery(client, events, sessions)
+      client.notify(new Error('noooo'))
+
+      expect(events.length).toEqual(1)
+      expect(events[0]._user.id).toBe(undefined)
+
+      client.startSession()
+
+      expect(sessions).toHaveLength(1)
+      expect(sessions[0].getUser().id).toBe(undefined)
+    })
+
+    it('should set device.id as user.id when collectUserIp=false', () => {
+      const client = new Client(
+        { apiKey: 'API_KEY_YEAH', collectUserIp: false },
+        {
+          ...schema,
+          collectUserIp: {
+            defaultValue: () => true,
+            validate: () => true,
+            message: ''
+          }
+        },
+        [plugin(navigator)]
+      )
+      const events: EventWithInternals[] = []
+      const sessions: SessionWithDevice[] = []
+
+      expect(client._cbs.e).toHaveLength(1)
+
+      mockDelivery(client, events, sessions)
+      client.notify(new Error('noooo'))
+
+      expect(events.length).toEqual(1)
+      expect(events[0]._user.id).toBe(events[0].device.id)
+
+      client.startSession()
+
+      expect(sessions).toHaveLength(1)
+      expect(sessions[0].getUser().id).toBe(events[0].device.id)
```

Comment:
Might make sense to check the IDs aren't undefined as well, otherwise this will pass if neither ID is present:

```suggestion
      expect(events).toHaveLength(1)
      expect(events[0]._user.id).toBeTruthy()
      expect(events[0]._user.id).toBe(events[0].device.id)

      client.startSession()

      expect(sessions).toHaveLength(1)
      expect(sessions[0].getUser().id).toBeTruthy()
      expect(sessions[0].getUser().id).toBe(events[0].device.id)
```